imap_processing/ultra/l1b/ultra_l1b_annotated.py

```diff
@@ -0,0 +1,50 @@
+"""Calculates Annotated Events for ULTRA L1b."""
+
+import typing
+
+import numpy as np
+import spiceypy as spice
+from numpy.typing import NDArray
+
+from imap_processing.spice.kernels import ensure_spice
+
+
+@ensure_spice
+@typing.no_type_check
+def get_particle_velocity(
```

Comment:
I'm think that this function may be useful for other instruments. Would you consider moving this into the `spice/geometry.py` module, renaming it to reflect that it is really just rotating a vector from one reference frame to another, and allowing for any `from_frame` and `to_frame`? Then you could add a simple wrapper function that is specifically useful for rotating from a specified instrument frame into the IMAP_DPS frame.

Reply:
I might walk this comment back. Maybe what we should do is create a ticket that is for adding the SPICE frame transform code in one PR and then modify this function once that is in place to use that function. There is something that I can pull from EMUS that is close to this implementation.

Reply:
Reviewed your code and I agree. If ok, I'll leave it for now, but create a ticket to modify this function